codespan-reporting/src/term/renderer.rs

```diff
@@ -42,7 +42,7 @@ pub enum MultiLabel<'diagnostic> {
     /// ```text
     /// │
     /// ```
-    Left,
+    Left(bool),
```

Comment:
What is the `bool` meant to be here? Would be handy to have it explained in the doc comment. Boolean flags can be a little hard to figure out in general!

Reply:
This bool indicates if this left label is "optional". The optionality is stored in the label. Then we can look at all the computed labels afterwards and see easily if the line can be skipped.

The previous approach of looking this up while looking at a single label would not work if there are mulitple `MultiLabel`s on one line which can both be skipped.

Reply:
I suppose the solution is to add a comment. Making it struct-like is probably too much.

Reply:
Riiiight, so we make it `true` if it's in the three lines at the start, or one line of context at the end?

Reply:
And otherwise `false` (it could overlap with another mandatory line)

Reply:
After some thought, I'm wondering if it would be cleaner/less confusing if we put this flag on the `Line` struct in the view. And we update it as we add new labels in the initial pass over the diagnostic.